Jenkinsfile

```diff
@@ -109,43 +139,130 @@ pipeline {
                description: 'Label to use for 9 VM functional tests')
-        string(name: 'CI_NVME_5_LABEL',
+        string(name: 'FUNCTIONAL_HARDWARE_MEDIUM_LABEL',
                defaultValue: 'ci_nvme5',
-               description: 'Label to use for 5 node Functional Hardware Medium stage')
-        string(name: 'CI_NVME_9_LABEL',
+               description: 'Label to use for the Functional Hardware Medium stage')
+        string(name: 'FUNCTIONAL_HARDWARE_MEDIUM_MD_ON_SSD_LABEL',
+               defaultValue: 'ci_nvme5',
+               description: 'Label to use for the Functional Hardware Medium MD on SSD stage')
+        string(name: 'FUNCTIONAL_HARDWARE_LARGE_LABEL',
+               defaultValue: 'ci_nvme9',
+               description: 'Label to use for 9 node Functional Hardware Large tests')
+        string(name: 'FUNCTIONAL_HARDWARE_LARGE_MD_ON_SSD_LABEL',
                defaultValue: 'ci_nvme9',
-               description: 'Label to use for 9 node Functional Hardware Large stage')
+               description: 'Label to use for the Functional Hardware Large MD on SSD stage')
+        string(name: 'CI_STORAGE_PREP_LABEL',
+               defaultValue: '',
+               description: 'Label for cluster to do a DAOS Storage Preparation')
+        string(name: 'CI_PROVISIONING_POOL',
+               defaultValue: '',
+               description: 'The pool of images to provision test nodes from')
         string(name: 'CI_BUILD_DESCRIPTION',
                defaultValue: '',
                description: 'A description of the build')
     }
 
     stages {
+        stage('Set Description') {
+            steps {
+                script {
+                    if (params.CI_BUILD_DESCRIPTION) {
+                        buildDescription params.CI_BUILD_DESCRIPTION
+                    }
+                }
+            }
+        }
         stage('Get Commit Message') {
             steps {
                 script {
                     env.COMMIT_MESSAGE = sh(script: 'git show -s --format=%B',
                                             returnStdout: true).trim()
+                    Map pragmas = [:]
+                    // can't use eachLine() here: https://issues.jenkins.io/browse/JENKINS-46988/
+                    env.COMMIT_MESSAGE.split('\n').each { line ->
+                        String key, value
+                        try {
+                            (key, value) = line.split(':', 2)
+                            if (key.contains(' ')) {
+                                return
+                            }
+                            pragmas[key.toLowerCase()] = value
+                        /* groovylint-disable-next-line CatchArrayIndexOutOfBoundsException */
+                        } catch (ArrayIndexOutOfBoundsException ignored) {
+                            // ignore and move on to the next line
+                        }
+                    }
+                    env.pragmas = pragmas
```

Comment:
Nice to see this being added.  Really cuts down on the stage's step noise by quite a bit.

Reply:
I'm curious how it cuts down on noise? (I'm misunderstanding something!)

Reply:
It loads the commit pragmas into an environment variable that can be queried/parsed without adding steps where we have to run `sh("git show"…)` every time we want to query/parse commit pragmas.